### ISSUE TITLE
The POM produced by gradlew createPom contains outdated, incorrect information

### ISSUE THREAD

**Issue by iambmelt** [AzureAD/azure-activedirectory-library-for-android#1033]
Output POM

```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <modelVersion>4.0.0</modelVersion>
  <groupId>com.microsoft.aad</groupId>
  <artifactId>adal</artifactId>
  <version>1.13.2</version>
  <packaging>aar</packaging>
  <name>adal</name>
  <description>Azure active directory library for Android gives you the ability to add Windows Azure Active Directory authentication to your application with just a few lines of additional code. Using our ADAL SDKs you can quickly and easily extend your existing application to all the employees that use Windows Azure AD and Active Directory on-premises using Active Directory Federation Services, including Office365 customers.</description>
  <url>https://github.com/MSOpenTech/azure-activedirectory-library-for-android</url>
  <inceptionYear>2014</inceptionYear>
  <licenses>
    <license>
      <name>MIT License</name>
    </license>
  </licenses>
  <developers>
    <developer>
      <id>microsoft</id>
      <name>Microsoft</name>
    </developer>
  </developers>
  <scm>
    <url>https://github.com/MSOpenTech/azure-activedirectory-library-for-android/tree/master</url>
  </scm>
  <properties>
    <branch>master</branch>
    <adalVersion>1.13.2</adalVersion>
  </properties>
  <dependencies>
    <dependency>
      <groupId>com.android.support</groupId>
      <artifactId>appcompat-v7</artifactId>
      <version>25.3.1</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.google.code.gson</groupId>
      <artifactId>gson</artifactId>
      <version>2.8.1</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.android.support</groupId>
      <artifactId>support-annotations</artifactId>
      <version>25.3.1</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.android.support</groupId>
      <artifactId>support-v4</artifactId>
      <version>25.3.1</version>
      <scope>compile</scope>
    </dependency>
  </dependencies>
</project>
```

- Declared `scm` is incorrect
- Project URL is incorrect (No longer MSOpenTech)

**Comment by iambmelt** [comment]
Experimental changes using `maven-publish`
```groovy
model {
    tasks.generatePomFileForAdalPublication {
        destination = file("$buildDir/generated-pom.xml")
    }
}
```
yields
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <modelVersion>4.0.0</modelVersion>
  <groupId>com.microsoft.aad</groupId>
  <artifactId>adal</artifactId>
  <version>1.13.2</version>
  <packaging>aar</packaging>
  <dependencies>
    <dependency>
      <groupId>com.android.support</groupId>
      <artifactId>appcompat-v7</artifactId>
      <version>25.3.1</version>
    </dependency>
    <dependency>
      <groupId>com.google.code.gson</groupId>
      <artifactId>gson</artifactId>
      <version>2.8.1</version>
    </dependency>
    <dependency>
      <groupId>com.android.support</groupId>
      <artifactId>support-annotations</artifactId>
      <version>25.3.1</version>
    </dependency>
    <dependency>
      <groupId>com.android.support</groupId>
      <artifactId>support-v4</artifactId>
      <version>25.3.1</version>
    </dependency>
    <dependency>
      <groupId>azure-activedirectory-library-for-android</groupId>
      <artifactId>common</artifactId>
      <version>unspecified</version>
    </dependency>
  </dependencies>
</project>

```

**Comment by iambmelt** [comment]
Need to figure out the proper POM to generate if a project is using `git submodule` -- can a `maven-exec` be used to do the checkout?

```xml
<plugin>
    <groupId>org.codehaus.mojo</groupId>
    <artifactId>exec-maven-plugin</artifactId>
    <inherited>false</inherited> <!-- only execute these in the parent -->
    <executions>
        <execution>
            <id>git submodule update</id>
            <phase>initialize</phase>
            <configuration>
                <executable>git</executable>
                <arguments>
                    <argument>submodule</argument>
                    <argument>update</argument>
                    <argument>--init</argument>
                    <argument>--recursive</argument>
                </arguments>
            </configuration>
            <goals>
                <goal>exec</goal>
            </goals>
        </execution>
    </executions>
</plugin>
```

**Comment by iambmelt** [comment]
closed by #1218
